cmd/password_test.go

```diff
@@ -0,0 +1,155 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/evcc-io/evcc/api/globalconfig"
+	"github.com/evcc-io/evcc/core/keys"
+	"github.com/evcc-io/evcc/server/db/settings"
+	"github.com/evcc-io/evcc/util/auth"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestPasswordCommandShouldNotUseConfigureEnvironment demonstrates why password
+// commands should use configureDatabase instead of configureEnvironment.
+// This test documents that configureEnvironment is unnecessarily slow and complex
+// for password management operations.
+func TestPasswordCommandShouldNotUseConfigureEnvironment(t *testing.T) {
+	t.Skip("This test documents the issue but is skipped to avoid slow test runs")
+
+	tmpDir := t.TempDir()
+	dbPath := filepath.Join(tmpDir, "test.db")
+	configPath := filepath.Join(tmpDir, "evcc.yaml")
+
+	// Create config with invalid sponsor token
+	configContent := `
+sponsortoken: invalid-token-that-will-fail-validation
+`
+	require.NoError(t, os.WriteFile(configPath, []byte(configContent), 0o644))
+
+	// Change to tmpDir to avoid file system issues with locale
+	oldWd, _ := os.Getwd()
+	os.Chdir(tmpDir)
+	defer os.Chdir(oldWd)
+
+	// Create i18n directory to prevent panic
+	require.NoError(t, os.MkdirAll("i18n", 0o755))
+
+	// Reset global config
+	conf = globalconfig.All{
+		Interval:     10 * time.Second,
+		Log:          "info",
+		SponsorToken: "invalid-token-that-will-fail-validation",
+		Network: globalconfig.Network{
+			Schema: "http",
+			Host:   "evcc.local",
+			Port:   7070,
+		},
+		Mqtt: globalconfig.Mqtt{
+			Topic: "evcc",
+		},
+		Database: globalconfig.DB{
+			Type: "sqlite",
+			Dsn:  dbPath,
+		},
+	}
+
+	// Set config file location
+	viper.SetConfigFile(configPath)
+
+	// Create a mock command
+	cmd := &cobra.Command{}
+	cmd.Flags().String(flagIgnoreDatabase, "", "")
+	cmd.Flags().String(flagHeaders, "", "")
+	cmd.PersistentFlags().String(flagTemplate, "", "")
+	cmd.PersistentFlags().String(flagTemplateType, "", "")
+
+	// Load config
+	err := loadConfigFile(&conf, true)
+	require.NoError(t, err)
+
+	// This is what the OLD code does - it calls configureEnvironment
+	// This is slow (~8 seconds) and unnecessary for password management
+	start := time.Now()
+	err = configureEnvironment(cmd, &conf)
+	elapsed := time.Since(start)
+
+	// configureEnvironment doesn't fail hard, but it's slow
+	t.Logf("configureEnvironment took %v (should be fast for password commands)", elapsed)
+	assert.Greater(t, elapsed.Seconds(), 5.0, "configureEnvironment is unnecessarily slow")
+}
+
+// TestPasswordSetWithDatabaseOnlySetup verifies that password management
+// works when using configureDatabase directly, even with invalid sponsor tokens.
+// This test passes on BOTH branches because it uses the correct approach.
+func TestPasswordSetWithDatabaseOnlySetup(t *testing.T) {
```

Comment:
these tests dont really test the command line request and are not trivial. Are they really valuable or can we simplify?

Reply:
Yes, lets make this shorter on only shorter and only test that the command succeeds even with a broken token. Also not a fan of the timing check.